physical/cache.go

```diff
@@ -34,16 +43,58 @@ func NewCache(b Backend, size int, logger log.Logger) *Cache {
 	c := &Cache{
 		backend: b,
 		lru:     cache,
+		locks:   make(map[string]*sync.RWMutex, 256),
+		logger:  logger,
 	}
+	if err := locksutil.CreateLocks(c.locks, 256); err != nil {
+		logger.Error("physical/cache: error creating locks", "error", err)
+		return nil
+	}
+
+	if txnl, ok := c.backend.(Transactional); ok {
+		c.transactional = txnl
+	}
+
 	return c
 }
 
+func (c *Cache) lockHashForKey(key string) string {
+	hf := sha1.New()
```

Comment:
Do we desire randomness to the keys to choose the locks? If not, encoding to hex should do the job. No?